android/java/org/chromium/chrome/browser/app/domain/NetworkModel.java

```diff
@@ -0,0 +1,53 @@
+package org.chromium.chrome.browser.app.domain;
+
+import android.text.TextUtils;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MediatorLiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import org.chromium.brave_wallet.mojom.BraveWalletConstants;
+import org.chromium.brave_wallet.mojom.JsonRpcService;
+import org.chromium.brave_wallet.mojom.JsonRpcServiceObserver;
+import org.chromium.mojo.system.MojoException;
+
+public class NetworkModel implements JsonRpcServiceObserver {
+    private final JsonRpcService mJsonRpcService;
+    private final MutableLiveData<String> _mChainId;
+    public LiveData<String> mChainId;
+    private final CryptoSharedData mSharedData;
+
+    public NetworkModel(JsonRpcService jsonRpcService, CryptoSharedData sharedData) {
+        mJsonRpcService = jsonRpcService;
+        mSharedData = sharedData;
+        _mChainId = new MutableLiveData<>(BraveWalletConstants.MAINNET_CHAIN_ID);
+        mChainId = _mChainId;
+        jsonRpcService.addObserver(this);
+    }
+
+    public void init() {
+        mJsonRpcService.getChainId(mSharedData.getCoinType(), chainId -> {
+            String id =
+                    TextUtils.isEmpty(chainId) ? BraveWalletConstants.MAINNET_CHAIN_ID : chainId;
+            _mChainId.postValue(id);
+            mJsonRpcService.setNetwork(id, mSharedData.getCoinType(), hasSetNetwork -> {});
```

Comment:
why do we need to set it here inside get? Maybe we do it only `if (!chainId.equals(id))`

Reply:
It is to handle a scenario when chainId is empty then we either set the mainnet otherwise use the existing and update the current one blindly, something I picked up from ios repo.

Reply:
done [6e727df](https://github.com/brave/brave-core/pull/13359/commits/6e727dfe19539650c41b8aa82985377799ca7f60)